src/category_theory/monoidal/braided.lean

```diff
@@ -283,4 +298,265 @@ def discrete.braided_functor (F : M →* N) : braided_functor (discrete M) (disc
+end
+
+lemma tensor_left_unitality (X₁ X₂ : C) :
+    (λ_ (X₁ ⊗ X₂)).hom
+  = ((λ_ (𝟙_ C)).inv ⊗ 𝟙 (X₁ ⊗ X₂)) ≫
+    tensor_μ C (𝟙_ C, 𝟙_ C) (X₁, X₂) ≫
+    ((λ_ X₁).hom ⊗ (λ_ X₂).hom) :=
+begin
+  dsimp [tensor_μ],
+  have :
+      ((λ_ (𝟙_ C)).inv ⊗ 𝟙 (X₁ ⊗ X₂)) ≫
+      (α_ (𝟙_ C) (𝟙_ C) (X₁ ⊗ X₂)).hom ≫
+      (𝟙 (𝟙_ C) ⊗ (α_ (𝟙_ C) X₁ X₂).inv)
+    = 𝟙 (𝟙_ C) ⊗ ((λ_ X₁).inv ⊗ 𝟙 X₂) := by pure_coherence,
+  slice_rhs 1 3 { rw this }, clear this,
+  slice_rhs 1 2 { rw [←tensor_comp, ←tensor_comp,
+                      comp_id, comp_id,
+                      left_unitor_inv_braiding] },
+  simp only [assoc],
+  have :
+      (𝟙 (𝟙_ C) ⊗ ((ρ_ X₁).inv ⊗ 𝟙 X₂)) ≫
+      (𝟙 (𝟙_ C) ⊗ (α_ X₁ (𝟙_ C) X₂).hom) ≫
+      (α_ (𝟙_ C) X₁ (𝟙_ C ⊗ X₂)).inv ≫
+      ((λ_ X₁).hom ⊗ (λ_ X₂).hom)
+    = (λ_ (X₁ ⊗ X₂)).hom := by pure_coherence,
+  rw this,
+end
+
+lemma tensor_right_unitality (X₁ X₂ : C) :
+    (ρ_ (X₁ ⊗ X₂)).hom
+  = (𝟙 (X₁ ⊗ X₂) ⊗ (λ_ (𝟙_ C)).inv) ≫
+    tensor_μ C (X₁, X₂) (𝟙_ C, 𝟙_ C) ≫
+    ((ρ_ X₁).hom ⊗ (ρ_ X₂).hom) :=
+begin
+  dsimp [tensor_μ],
+  have :
+      (𝟙 (X₁ ⊗ X₂) ⊗ (λ_ (𝟙_ C)).inv) ≫
+      (α_ X₁ X₂ (𝟙_ C ⊗ 𝟙_ C)).hom ≫
+      (𝟙 X₁ ⊗ (α_ X₂ (𝟙_ C) (𝟙_ C)).inv)
+    = (α_ X₁ X₂ (𝟙_ C)).hom ≫
+      (𝟙 X₁ ⊗ ((ρ_ X₂).inv ⊗ 𝟙 (𝟙_ C))) := by pure_coherence,
+  slice_rhs 1 3 { rw this }, clear this,
+  slice_rhs 2 3 { rw [←tensor_comp, ←tensor_comp,
+                      comp_id, comp_id,
+                      right_unitor_inv_braiding] },
+  simp only [assoc],
+  have :
+      (α_ X₁ X₂ (𝟙_ C)).hom ≫
+      (𝟙 X₁ ⊗ ((λ_ X₂).inv ⊗ 𝟙 (𝟙_ C))) ≫
+      (𝟙 X₁ ⊗ (α_ (𝟙_ C) X₂ (𝟙_ C)).hom) ≫
+      (α_ X₁ (𝟙_ C) (X₂ ⊗ 𝟙_ C)).inv ≫
+      ((ρ_ X₁).hom ⊗ (ρ_ X₂).hom)
+    = (ρ_ (X₁ ⊗ X₂)).hom := by pure_coherence,
+  rw this,
+end
+
+/-
+Diagram B6 from Proposition 1 of André Joyal and Ross Street,
+"Braided monoidal categories", Macquarie Math Reports 860081 (1986).
+-/
+lemma tensor_associativity_aux (W X Y Z : C) :
+    ((β_ W X).hom ⊗ 𝟙 (Y ⊗ Z)) ≫
+    (α_ X W (Y ⊗ Z)).hom ≫
+    (𝟙 X ⊗ (α_ W Y Z).inv) ≫
+    (𝟙 X ⊗ (β_ (W ⊗ Y) Z).hom) ≫
+    (𝟙 X ⊗ (α_ Z W Y).inv)
+  = (𝟙 (W ⊗ X) ⊗ (β_ Y Z).hom) ≫
+    (α_ (W ⊗ X) Z Y).inv ≫
+    ((α_ W X Z).hom ⊗ 𝟙 Y) ≫
+    ((β_ W (X ⊗ Z)).hom ⊗ 𝟙 Y) ≫
+    ((α_ X Z W).hom ⊗ 𝟙 Y) ≫
+    (α_ X (Z ⊗ W) Y).hom :=
+begin
+  slice_rhs 3 5 { rw [←tensor_comp, ←tensor_comp,
+                      hexagon_forward,
+                      tensor_comp, tensor_comp] },
+  slice_rhs 5 6 { rw [associator_naturality] },
+  slice_rhs 2 3 { rw [←associator_inv_naturality] },
+  slice_rhs 3 5 { rw [←pentagon_hom_inv] },
+  slice_rhs 1 2 { rw [tensor_id,
+                      id_tensor_comp_tensor_id,
+                      ←tensor_id_comp_id_tensor] },
+  slice_rhs 2 3 { rw [← tensor_id, associator_naturality] },
+  slice_rhs 3 5 { rw [←tensor_comp, ←tensor_comp,
+                      ←hexagon_reverse,
+                      tensor_comp, tensor_comp] },
+end
+
+set_option class.instance_max_depth 64
```

Comment:
I had to add this because `pure_coherence` was failing to prove the goals in `tensor_associativity`, and the error said the type checker reached the maximum instance depth, so I tried to double it.  What's the right way to handle this kind of issue?

Reply:
Hmm, it's unfortunate that this is showing up in real-world examples. Perhaps it is a sign that the typeclass approach is not the best one, and we should go back to @yuma-mizuno's approach. Not sure.

In the meantime I have added a comment to the doc-string for `coherence` which explains exactly this solution.

Reply:
The coherence tactic internally increasing the typeclass search depth now; can you try removing this?

Reply:
Sadly, `coherence` fails to finish the proof of `tensor_associativity` when I remove this.